WordPress/gutenberg
# Image: Add the ability for a plugin to disable Image Editor

## Description

This adds the ability for a plugin to disable the new Image Editor tools, so if it is a feature they do not want to offer, or does not work then it can be disabled programmatically.

Uses the block-editor settings store to disable.

## How has this been tested?

1. Apply PR and confirm no change, everything works as expected (default does nothing different)

2. Try disabling, add a simple plugin that includes something like the following and confirm the new image editing tools are disable as expected. Additionally confirm other image features still work as expected.

```php
add_filter( 'block_editor_settings', function( $settings ) {
	$settings['imageEditing'] = false;
	return $settings;
} );
```

## Types of changes

Disables Image Editing tools by checking if block-editor settings has been set to false.


## Checklist:
